addons/phone_validation/models/mail_thread_phone.py

```diff
@@ -33,7 +33,11 @@ class PhoneMixin(models.AbstractModel):
     phone_blacklisted = fields.Boolean(
         string='Phone Blacklisted', compute="_compute_phone_blacklisted", compute_sudo=True, store=False,
         search="_search_phone_blacklisted", groups="base.group_user",
-        help="If the email address is on the blacklist, the contact won't receive mass mailing anymore, from any list")
+        help="If the phone number is on the blacklist, the contact won't receive mass mailing sms anymore, from any list")
+    mobile_blacklisted = fields.Boolean(
```

Comment:
I understand there is an issue with number of phone-like fields being variable. I think this implementation is not very clear as we have phone_blacklisted that means "is first found sanitized number based on all phone fields blacklisted" and mobile_blacklisted that means "is mobile field blacklisted".

Looking at the code it seems the use case is: we could have a phone_sanitized that is not blacklisted (phone is valid and not blacklisted) while having mobile blacklisted.

Existing use cases: we have 1 or 2 phone fields (generally mobile, mobile + phone, and a mobile_phone strange field). Why not having, even if it is quite ugly

  * phone_field1_blacklisted = fields.Boolean()
  * phone_field2_blacklisted = fields.Boolean()

That are computed based on _phone_get_number_fields, in the same order. That way we know which one is linked to a given model field. If there is no second phone field, just to it to False.

That way even if it is a bit hackish we can really say which number is blacklisted.

What do you think ?

Reply:
@tde-banana-odoo Your solution seems more generic and adapted to the current mixin.

However, I wonder how we're going to use it in views?
Are we going to assume that mobile is linked to `phone_field1_blacklisted` ?
And phone is linked to `phone_field2_blacklisted` ?
(based on what we implemented in "_phone_get_number_fields" for the related model)

Which will look something like:
`<field name="mobile" /><field name="phone_field1_blacklisted" widget="phone_blacklist" />`
Seems a bit cryptic.

And one more thing:
Even though we will have 2 "x_blacklisted" fields, we will not be able to compute both of them correctly, right?
Examples:
- if phone_field2_blacklisted is True, then we know for sure that phone_field1_blacklisted is False and correct
- if phone_field1_blacklisted is True, then phone_field2_blacklisted will be False **but the number could be blacklisted as well**, and we have no way to know
(I'm aware we have the same problem with the current "mobile_blacklisted" implementation, just wanted to know if you had ideas for that).

Side note: it's way to early for this, my head hurts.

Reply:
My proposal is that field1 is the first field defined in the _phone_get_number_fields, field2 the second one. Its blacklist state is based on "is sanitized version of field blaklisted". So it follows the order of field declaration. So fieldX_blacklisted is based only on actual value of field.

Another way round would be to say
  * mobile_blacklisted is looking for mobile or mobile_phone (stock picking) or mobile_nbr (tests) field or any version of x_thosefields if present in phone_get_nbr_field
  * phone_blacklisted is looking for phone or phone_nbr (tests) field or any version of x_thosefields if present in phone_get_nbr_field
  * rename current phone_blacklisted into phone_sanitized_blacklisted

That way it is clearer and normally works for all current use cases in odoo.

Better version would be to separate completely computation (phone_sanitized, phone_blacklisted, mobile_sanitized, mobile_blacklisted) but it is more complex as it has an impact on sms sending, mass sms, ... maybe for the next step.

Reply:
I was about to suggest that renaming too.

It won't work if you want to add your "brol_number" field that goes into "_phone_get_number_fields", but it's compliant with all our current needs and may help understanding what's going on in views as well.

It still doesn't solve the problem that if mobile_blacklisted is True, phone_blacklisted will always be False even though the phone number could actually be blacklisted.
But I guess we could consider it as a non-handled edge case.

I let you take the final decision on this since you're the mixin creator after all.

Reply:
Just to make sure I understood correctly. There should be 3 Booleans:
- `phone_sanitized_blacklisted`
- `phone_blacklisted`
- `mobile_blacklisted`
Where `phone_blacklisted` and `mobile_blacklisted` should look for all the different possible field name variants (e.g. mobile, mobile_phone, and mobile_nbr) to correctly calculate them. And then I determine whether or not to add the unblacklist button based on `phone_blacklisted` and mobile_blacklisted` values? And `phone_sanitized_blacklisted` won't be needed in the views?

Reply:
Ah I just went for it and added in those 3 booleans. One plus is with the new boolean, we can now have phone/mobile fields both have the "unblacklist button" show up next to them when it's the same number in both. I renamed all the instances of the former `phone_blacklisted` as `phone_sanitized_blacklisted` and updated some of the function names to better match the new naming scheme.

Reply:
Ah I just went for it and added in those 3 booleans. One plus is with the new boolean, we can now have phone/mobile fields both have the "unblacklist button" show up next to them when it's the same number in both. I renamed all the instances of the former `phone_blacklisted` as `phone_sanitized_blacklisted` and updated some of the function names to better match the new naming scheme.

Reply:
Indeed more fields but it is more explicit :) . I think that usability speaking it is better to clearly know which number is blacklisted...

Reply:
Agreed. Let me know if there's anything else I should do/change :)